flytekit/configuration/__init__.py

```diff
@@ -418,19 +425,31 @@ def auto(cls, config_file: typing.Optional[typing.Union[str, ConfigFile]] = None
             kwargs, "client_credentials_secret", _internal.Credentials.CLIENT_CREDENTIALS_SECRET.read(config_file)
         )
 
+        is_client_secret = False
         client_credentials_secret = read_file_if_exists(
             _internal.Credentials.CLIENT_CREDENTIALS_SECRET_LOCATION.read(config_file)
         )
-        if client_credentials_secret and client_credentials_secret.endswith("\n"):
-            logger.info("Newline stripped from client secret")
-            client_credentials_secret = client_credentials_secret.strip()
+        if client_credentials_secret:
+            is_client_secret = True
+            if client_credentials_secret.endswith("\n"):
+                logger.info("Newline stripped from client secret")
+                client_credentials_secret = client_credentials_secret.strip()
         kwargs = set_if_exists(
             kwargs,
             "client_credentials_secret",
             client_credentials_secret,
         )
+
+        client_credentials_secret = read_env_var_if_exists(
```

Comment:
can we just make this a one-liner and remove the helper?

Reply:
thanks for review!
i wonder if this `:=` this operator is usable or not (supported with python 3.8)
or i might change like following:
```python
if _internal.Credentials.CLIENT_CREDENTIALS_SECRET_ENV_VAR.read(config_file):
    client_credentials_secret = os.getenv(_internal.Credentials.CLIENT_CREDENTIALS_SECRET_ENV_VAR.read(config_file))
```
or create a variable for it